python-package/lightgbm/sklearn.py

```diff
@@ -451,34 +426,26 @@ def apply(self, X, num_iteration=0):
         -------
         X_leaves : array_like, shape=[n_samples, n_trees]
         """
-        return self._Booster.predict(X,
-                                     pred_leaf=True,
-                                     num_iteration=num_iteration)
+        return self.booster_.predict(X, pred_leaf=True, num_iteration=num_iteration)
 
-    def evals_result(self):
-        """
-        Return the evaluation results.
-
-        Returns
-        -------
-        evals_result : dictionary
-        """
-        if self.evals_result_:
-            evals_result = self.evals_result_
-        else:
-            raise LightGBMError('No results found.')
-
-        return evals_result
-
-    def feature_importance(self):
-        """
-        Feature importances
+    @property
+    def booster_(self):
```

Comment:
It's difficult to decide to use property or method.
The key is whether we allow users to set `self._Booster`.
I guess `self._Booster` is a **private** member (because of the prefix of `_` ), therefore it's better not to provide any getter, setter, and deleter (i.e. have one as property).
(If it is a property, we may access it like `gbm.booster_ = another_Booster`)

Moreover, the sklearn interface of LightGBM is very similar to that of XGBoost, so if we don't have any strong reasons, it may be bettor to follow the way of XGBoost.
(cf. https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/sklearn.py#L153)